debugtools/DDR_VM/src/com/ibm/j9ddr/BytecodeGenerator.java

```diff
@@ -21,95 +21,1724 @@
+			case StructureTypeManager.TYPE_STRUCTURE:
+				doStructureMethod(field);
+				break;
+			case StructureTypeManager.TYPE_STRUCTURE_POINTER:
+				doStructurePointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_POINTER:
+				doPointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_ARRAY:
+				doArrayMethod(field);
+				break;
+			case StructureTypeManager.TYPE_J9SRP:
+				doSRPMethod(field, false);
+				break;
+			case StructureTypeManager.TYPE_J9WSRP:
+				doSRPMethod(field, true);
+				break;
+			case StructureTypeManager.TYPE_J9SRP_POINTER:
+				doSRPPointerMethod(field, false);
+				break;
+			case StructureTypeManager.TYPE_J9WSRP_POINTER:
+				doSRPPointerMethod(field, true);
+				break;
+			case StructureTypeManager.TYPE_FJ9OBJECT:
+				doFJ9ObjectMethod(field);
+				break;
+			case StructureTypeManager.TYPE_FJ9OBJECT_POINTER:
+				doFJ9ObjectPointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_J9OBJECTCLASS:
+				doJ9ObjectClassMethod(field);
+				break;
+			case StructureTypeManager.TYPE_J9OBJECTCLASS_POINTER:
+				doJ9ObjectClassPointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_J9OBJECTMONITOR:
+				doJ9ObjectMonitorMethod(field);
+				break;
+			case StructureTypeManager.TYPE_J9OBJECTMONITOR_POINTER:
+				doJ9ObjectMonitorPointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_VOID:
+				doStructureMethod(field);
+				break;
+			case StructureTypeManager.TYPE_BOOL:
+				doBooleanMethod(field);
+				break;
+			case StructureTypeManager.TYPE_DOUBLE:
+				doDoubleMethod(field);
+				break;
+			case StructureTypeManager.TYPE_FLOAT:
+				doFloatMethod(field);
+				break;
+			case StructureTypeManager.TYPE_ENUM:
+				doEnumMethod(field);
+				break;
+			case StructureTypeManager.TYPE_ENUM_POINTER:
+				doEnumPointerMethod(field);
+				break;
+			case StructureTypeManager.TYPE_BITFIELD:
+				int colonIndex = typeName.indexOf(':');
+				if (colonIndex <= 0) {
+					throw new IllegalArgumentException(String.format("%s.%s : %s is not a valid bitfield",
+							structure.getName(), fieldName, typeName));
+				}
+				String baseType = typeName.substring(0, colonIndex);
+				int width = Integer.parseInt(typeName.substring(colonIndex + 1));
+				doBitfieldMethod(field, baseType, width);
+				break;
+
+			default:
+				if ((StructureTypeManager.TYPE_SIMPLE_MIN <= type) && (type <= StructureTypeManager.TYPE_SIMPLE_MAX)) {
+					doSimpleTypeMethod(field, type);
+				} else {
+					// omit unhandled field type
+				}
+				break;
+			}
+		}
+	}
+
+	/* Sample generated code:
+	 *
+	 * public ObjectMonitorReferencePointer objectMonitorLookupCacheEA() throws CorruptDataException {
+	 *     return ObjectMonitorReferencePointer.cast(nonNullFieldEA(J9VMThread._objectMonitorLookupCacheOffset_));
+	 * }
+	 */
+	private void doArrayMethod(FieldDescriptor field) {
+		String fieldType = field.getType();
+		String componentType = fieldType.substring(0, fieldType.lastIndexOf('[')).trim();
```

Comment:
Assumes arrays are always single dimension.

Reply:
No. If the field type is `Whatever[4][5]` it will analyze `Whatever[4]` and notice the component type has type `TYPE_ARRAY`. This is the same as the behavior found in `PointerGenerator.java`.